db/hints/manager.hh

```diff
@@ -308,21 +310,33 @@ private:
     }
 
 public:
-    ep_managers_map_type::iterator find_ep_manager(endpoint_id ep_key) noexcept {
-        return _ep_managers.find(ep_key);
+    host_managers_map_type::iterator find_host_manager(locator::host_id hid) noexcept {
+        return _host_managers.find(hid);
     }
 
-    ep_managers_map_type::const_iterator find_ep_manager(endpoint_id ep_key) const noexcept {
-        return _ep_managers.find(ep_key);
+    host_managers_map_type::const_iterator find_host_manager(locator::host_id hid) const noexcept {
+        return _host_managers.find(hid);
     }
 
-    ep_managers_map_type::iterator ep_managers_end() noexcept {
-        return _ep_managers.end();
+    host_managers_map_type::iterator host_managers_end() noexcept {
+        return _host_managers.end();
     }
 
-    ep_managers_map_type::const_iterator ep_managers_end() const noexcept {
-        return _ep_managers.end();
+    host_managers_map_type::const_iterator host_managers_end() const noexcept {
+        return _host_managers.end();
     }
+
+private:
+    endpoint_id convert_host_id_to_ep(locator::host_id hid) const;
```

Comment:
The function with this signature can't be implemented. The `token_metadata` may change and this function would fail on left `host_id`-s.